### PR TITLE
feature/tenant-aware-pfdhcplistener

### LINKED ISSUE #5991: pfdhcplistener is not tenant aware
**Is your feature request related to a problem? Please describe.**
When you are in a multi-tenant setup and you use a DHCP sensor, `pfdhcplistener` will process DHCP packets and create/update all nodes in **default** tenant even if nodes already exist in other tenants.

**Describe the solution you'd like**
1. Update nodes already created in DB in correct tenant
1. Create node in correct tenant when node is unknown: we could create a mapping between subnets declared in DHCP and tenants

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Make the pfdhcplistener tenant aware

Impacts
=======
pfdhcplistener

NEWS file entries
=======================
Enhancements
-------------
* The pfdhcplistener is tenant aware.

Issue
=====
fixes #5991

Delete branch after merge
=========================
NO

Checklist
=========
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

UPGRADE file entries
=========
To add the tenant_id to all network configurations run 
perl /usr/local/pf/addons/upgrade/to-11.0-network-conf.pl

### CODE REVIEW

File: lib/pf/ConfigStore/L2Network.pm [hunk 1]
```diff
@@ -12,10 +12,7 @@ ConfigStore for L2 networks in networks.conf
 
 use Moo;
 use namespace::autoclean;
-use pf::log;
-use pf::config qw(%ConfigNetworks %Config);
-use pf::util qw(isenabled);
-use pf::file_paths qw($network_config_file);
+#use pf::constants::config;
```
Comment (extrafu) [inline lib/pf/ConfigStore/L2Network.pm]: Remove?

File: t/unittest/dhcp/processor_v4.t [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+BEGIN {
+    #include test libs
+    use lib qw(/usr/local/pf/t);
+    #Module for overriding configuration paths
+    use setup_test_config;
+}
+
+use Test::More tests => 5;
+
+#This test will running last
```
Comment (extrafu) [inline t/unittest/dhcp/processor_v4.t]: Rephrase

**Comment by jrouzierinverse** [comment]
done

**Comment by nqb** [comment]
I'm self-assigning in order to test.

File: addons/upgrade/to-11.0-network-conf.pl [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/perl
+
+=head1 NAME
+
+to-10.2-pfdetect-conf -
+
+=head1 DESCRIPTION
+
+Add the new field tenant_id to the detect parsers
+
```
Comment (nqb) [inline addons/upgrade/to-11.0-network-conf.pl]: Wrong description

**Review by nqb — CHANGES_REQUESTED** [review]

**Comment by jrouzierinverse** [comment]
Updated and rebased

**Comment by nqb** [comment]
@satkunas,

When I tried to create an unmanaged network using "New Routed Network" button, I don't have "Unmanaged" choice in dropdown list of "Type" field. I checked return of API (OPTIONS /api/v1/config/routed_networks) and it contains:

```
0: {text: "Isolation", value: "vlan-isolation"}
1: {text: "Registration", value: "vlan-registration"}
2: {text: "DNS Enforcement", value: "dns-enforcement"}
3: {text: "Inline Layer 3", value: "inlinel3"}
4: {text: "Unmanaged", value: "other"}
```

**Comment by nqb** [comment]
Closing this PR in favour of #6204

**Pull request closed without merging**